man/zram-generator.md

```diff
@@ -0,0 +1,57 @@
+zram-generator(8) -- Systemd unit generator for zram swap devices
+=================================================================
+
+## SYNOPSIS
+
+`/lib/systemd/system-generators/zram-generator` `TARGET_DIR` [*2RGET_DIR* *3RGET_DIR*]<br />
+`/lib/systemd/system-generators/zram-generator` --setup-device `DEVICE`<br />
+`/lib/systemd/system-generators/zram-generator` --reset-device `DEVICE`
+
+## DESCRIPTION
+
+`zram-generator` is a generator that creates zram devices and systemd units to configure, format, and enable swap on them.
+
+The generator will be invoked by systemd early at boot, the generator will then:
+
+  1. read configuration files from *{/etc,/lib}/systemd/zram-generator.conf[.d]* (see zram-generator.conf(5) for details);
+  2. if systemd-detect-virt(1) exists and returns non-*0*, exit, since continuing would mean changing the host system;
```

Comment:
Implementation detail... I'd prefer to say, outside of this list, that "The generator does nothing in containers, except when running in test mode, in particular when $ZRAM_GENERATOR_ROOT is set." (The distinction about containers is important, since it *does* work on VMs.)

Reply:
Oh, good catch! Updated